test/twitter_web/controllers/browse_posts_controller_test.exs

```diff
@@ -0,0 +1,32 @@
+defmodule TwitterWeb.BrowseControllerTest do
+  use TwitterWeb.ConnCase
+
+  # make post from one user account, then relogin -> should still
+  # see the post, as it is the public post controller
+  setup %{conn: conn} do
+    author = user_fixture()
+    reader = user_fixture()
+
+    # author writes post and logs out
+    conn = init_test_session(conn, current_user: author, user_id: author.id)
+    post = post_fixture(author, %{body: "i love you"})
```

Comment:
I feel like this can be shortened by calling your context functions here instead of init_test_session -> clear_session pattern. Would you mind attempting to do that instead to insert post into db?

Reply:
Ok, fixed that in other places too